packages/astro-integration/src/astro-bundle-analysis/astroBundleAnalysisPlugin.ts

```diff
@@ -0,0 +1,52 @@
+import {
+  type BundleAnalysisUploadPluginArgs,
+  type BundleAnalysisUploadPluginReturn,
+  red,
+} from "@codecov/bundler-plugin-core";
+import { getBundleName } from "./getBundleName";
+
+// @ts-expect-error this value is being replaced by rollup
+const PLUGIN_NAME = __PACKAGE_NAME__ as string;
+// @ts-expect-error this value is being replaced by rollup
+const PLUGIN_VERSION = __PACKAGE_VERSION__ as string;
+
+interface AstroBundleAnalysisArgs extends BundleAnalysisUploadPluginArgs {
+  target: "client" | "server";
+}
+
+type AstroBundleAnalysisPlugin = (
+  args: AstroBundleAnalysisArgs,
+) => BundleAnalysisUploadPluginReturn;
+
+export const astroBundleAnalysisPlugin: AstroBundleAnalysisPlugin = ({
+  output,
+  target,
+}) => ({
+  version: output.version,
+  name: PLUGIN_NAME,
+  pluginVersion: PLUGIN_VERSION,
+  vite: {
```

Comment:
We can cross this bridge if/when it comes, but will we end up with any tricky dependency issues where we expect a certain version of vite to be used with astro? I guess that is more just up to astro to maintain that compatibility